File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.java
import java.util.*;

class Solution {

    void swap(int[] nums, int i, int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }

    public List<Integer> findDisappearedNumbers(int[] nums) {
        int n = nums.length;
        List<Integer> result = new ArrayList<>();

        for (int i = 0; i < n;) {
            int correctIndex = nums[i] - 1; 
            if (nums[i] > 0 && nums[i] <= n && nums[i] != nums[correctIndex]) {
                swap(nums, i, correctIndex);  
            } else {
                i++;
            }
        }

        for (int i = 0; i < n; i++) {
            if (nums[i] != i + 1) result.add(i + 1);
        }

        return result;
    }
}

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        map={}

        for x in nums:
            if x in map:
                map[x]+=1
            else:
                map[x]=1
        
        n=len(nums)
        lst=[]
        for i in range(n):
            if i+1 not in map:
                lst.append(i+1)

        return lst 
        